Crash when launching app with latest update

Hi!

With the latest update, the app crashes as soon as it starts. If I switch back to the old library, it works as it should. I've tried to rebuild all, but still same problem.

My project was migrated to AndroidX some months ago, so that should not be any issue. Running on Android 7 on an emulator. (My project doesn't use any kotlin code, if that matters.) I'm using this library together with Calligraphy3.

Here is the stack trace:

```
java.lang.RuntimeException: Unable to start activity ComponentInfo{com.the.app/com.the.app.activities.MainActivity}: android.view.InflateException: Binary XML file line #24: Binary XML file line #24: Error inflating class LinearLayout
        at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2646)
        at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2707)
        at android.app.ActivityThread.-wrap12(ActivityThread.java)
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1460)
        at android.os.Handler.dispatchMessage(Handler.java:102)
        at android.os.Looper.loop(Looper.java:154)
        at android.app.ActivityThread.main(ActivityThread.java:6077)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:866)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:756)
     Caused by: android.view.InflateException: Binary XML file line #24: Binary XML file line #24: Error inflating class LinearLayout
     Caused by: android.view.InflateException: Binary XML file line #24: Error inflating class LinearLayout
     Caused by: java.lang.IllegalArgumentException: Parameter specified as non-null is null: method kotlin.jvm.internal.Intrinsics.checkParameterIsNotNull, parameter parent
        at io.github.inflationx.viewpump.ViewPumpLayoutInflater.onCreateView(ViewPumpLayoutInflater.kt)
        at android.view.LayoutInflater.createViewFromTag(LayoutInflater.java:785)
        at android.view.LayoutInflater.createViewFromTag(LayoutInflater.java:727)
        at android.view.LayoutInflater.inflate(LayoutInflater.java:495)
        at io.github.inflationx.viewpump.ViewPumpLayoutInflater.inflate(ViewPumpLayoutInflater.kt:54)
        at android.view.LayoutInflater.inflate(LayoutInflater.java:426)
        at io.github.inflationx.viewpump.ViewPumpLayoutInflater.inflate(ViewPumpLayoutInflater.kt:45)
        at android.view.LayoutInflater.inflate(LayoutInflater.java:377)
        at com.android.internal.policy.DecorView.onResourcesLoaded(DecorView.java:1776)
        at com.android.internal.policy.PhoneWindow.generateLayout(PhoneWindow.java:2613)
        at com.android.internal.policy.PhoneWindow.installDecor(PhoneWindow.java:2686)
        at com.android.internal.policy.PhoneWindow.getDecorView(PhoneWindow.java:2049)
        at androidx.appcompat.app.AppCompatDelegateImpl.createSubDecor(AppCompatDelegateImpl.java:575)
        at androidx.appcompat.app.AppCompatDelegateImpl.ensureSubDecor(AppCompatDelegateImpl.java:518)
        at androidx.appcompat.app.AppCompatDelegateImpl.setContentView(AppCompatDelegateImpl.java:466)
        at androidx.appcompat.app.AppCompatActivity.setContentView(AppCompatActivity.java:140)
        at com.the.app.activities.MainActivity.onCreate(MainActivity.java:270)
        at android.app.Activity.performCreate(Activity.java:6662)
        at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1118)
        at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2599)
        at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2707)
        at android.app.ActivityThread.-wrap12(ActivityThread.java)
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1460)
        at android.os.Handler.dispatchMessage(Handler.java:102)
        at android.os.Looper.loop(Looper.java:154)
        at android.app.ActivityThread.main(ActivityThread.java:6077)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:866)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:756)
```

--- a/viewpump/src/main/java/io/github/inflationx/viewpump/-ViewPumpActivityFactory.kt
+++ b/viewpump/src/main/java/io/github/inflationx/viewpump/-ViewPumpActivityFactory.kt
@@ -33,6 +33,6 @@ internal interface ViewPumpActivityFactory {
    * @return The view passed in, or null if nothing was passed in.
    * @see android.view.LayoutInflater.Factory2
    */
-  fun onActivityCreateView(parent: View, view: View, name: String, context: Context,
-      attrs: AttributeSet): View?
+  fun onActivityCreateView(parent: View?, view: View, name: String, context: Context,
+      attrs: AttributeSet?): View?
 }

--- a/viewpump/src/main/java/io/github/inflationx/viewpump/ViewPumpContextWrapper.kt
+++ b/viewpump/src/main/java/io/github/inflationx/viewpump/ViewPumpContextWrapper.kt
@@ -77,8 +77,8 @@ class ViewPumpContextWrapper private constructor(base: Context) : ContextWrapper
      * @return The same view passed in, or null if null passed in.
      */
     @JvmStatic
-    fun onActivityCreateView(activity: Activity, parent: View, view: View, name: String,
-        context: Context, attr: AttributeSet): View? {
+    fun onActivityCreateView(activity: Activity, parent: View?, view: View, name: String,
+        context: Context, attr: AttributeSet?): View? {
       return get(activity).onActivityCreateView(parent, view, name, context, attr)
     }
 

--- a/viewpump/src/main/java/io/github/inflationx/viewpump/ViewPumpLayoutInflater.kt
+++ b/viewpump/src/main/java/io/github/inflationx/viewpump/ViewPumpLayoutInflater.kt
@@ -118,8 +118,8 @@ internal class ViewPumpLayoutInflater : LayoutInflater, ViewPumpActivityFactory 
    * The Activity onCreateView (PrivateFactory) is the third port of call for LayoutInflation.
    * We opted to manual injection over aggressive reflection, this should be less fragile.
    */
-  override fun onActivityCreateView(parent: View, view: View, name: String, context: Context,
-      attrs: AttributeSet): View? {
+  override fun onActivityCreateView(parent: View?, view: View, name: String, context: Context,
+      attrs: AttributeSet?): View? {
     return ViewPump.get()
         .inflate(InflateRequest(
             name = name,
@@ -136,7 +136,7 @@ internal class ViewPumpLayoutInflater : LayoutInflater, ViewPumpActivityFactory 
    * BUT only for none CustomViews.
    */
   @Throws(ClassNotFoundException::class)
-  override fun onCreateView(parent: View, name: String, attrs: AttributeSet): View? {
+  override fun onCreateView(parent: View?, name: String, attrs: AttributeSet?): View? {
     return ViewPump.get()
         .inflate(InflateRequest(
             name = name,
@@ -154,7 +154,7 @@ internal class ViewPumpLayoutInflater : LayoutInflater, ViewPumpActivityFactory 
    * Basically if this method doesn't inflate the View nothing probably will.
    */
   @Throws(ClassNotFoundException::class)
-  override fun onCreateView(name: String, attrs: AttributeSet): View? {
+  override fun onCreateView(name: String, attrs: AttributeSet?): View? {
     return ViewPump.get()
         .inflate(InflateRequest(
             name = name,
@@ -283,7 +283,7 @@ internal class ViewPumpLayoutInflater : LayoutInflater, ViewPumpActivityFactory 
 
     private val viewCreator: FallbackViewCreator = WrapperFactoryViewCreator(factory)
 
-    override fun onCreateView(name: String, context: Context, attrs: AttributeSet): View? {
+    override fun onCreateView(name: String, context: Context, attrs: AttributeSet?): View? {
       return ViewPump.get()
           .inflate(InflateRequest(
               name = name,
@@ -310,12 +310,12 @@ internal class ViewPumpLayoutInflater : LayoutInflater, ViewPumpActivityFactory 
   private open class WrapperFactory2(factory2: LayoutInflater.Factory2) : LayoutInflater.Factory2 {
     private val viewCreator = WrapperFactory2ViewCreator(factory2)
 
-    override fun onCreateView(name: String, context: Context, attrs: AttributeSet): View? {
+    override fun onCreateView(name: String, context: Context, attrs: AttributeSet?): View? {
       return onCreateView(null, name, context, attrs)
     }
 
     override fun onCreateView(parent: View?, name: String, context: Context,
-        attrs: AttributeSet): View? {
+        attrs: AttributeSet?): View? {
       return ViewPump.get()
           .inflate(InflateRequest(
               name = name,
@@ -346,7 +346,7 @@ internal class ViewPumpLayoutInflater : LayoutInflater, ViewPumpActivityFactory 
     private val viewCreator = PrivateWrapperFactory2ViewCreator(factory2, inflater)
 
     override fun onCreateView(parent: View?, name: String, context: Context,
-        attrs: AttributeSet): View? {
+        attrs: AttributeSet?): View? {
       return ViewPump.get()
           .inflate(InflateRequest(
               name = name,
@@ -374,7 +374,7 @@ internal class ViewPumpLayoutInflater : LayoutInflater, ViewPumpActivityFactory 
 
     private val CLASS_PREFIX_LIST = setOf("android.widget.", "android.webkit.")
     private val CONSTRUCTOR_ARGS_FIELD: Field by lazy {
-      requireNotNull(LayoutInflater::class.java.getDeclaredField("CONSTRUCTOR_ARGS_FIELD")) {
+      requireNotNull(LayoutInflater::class.java.getDeclaredField("mConstructorArgs")) {
         "No constructor arguments field found in LayoutInflater!"
       }.apply { isAccessible = true }
     }

